### DISCUSSION TITLE
Parsing GQL -> TS enum member descriptions

### DISCUSSION THREAD

**Discussion by mccauleyp** [dotansimha/graphql-code-generator#8748] [Q&A]
I have some GraphQL enums where the values are used as menu items in a drop down menu and the descriptions are used as tooltip text to provide information about each item. 

Currently I'm running a `__type` query when the page loads, but I'd like to instead use the generated TypeScript types so that this query isn't needed and so that introspection queries can be disabled generally. The GraphQL enums converted to TS look like:

```
export enum MyEnum {
  /** Member 1 description */
  Member1 = 'MEMBER1',
  /** Member 1 description */
  Member2 = 'MEMBER2',
}
```

I'm looking for advice on how to pull out the description fields. Is there a way to do this with `graphql-code-generator` that perhaps maps GraphQL enums to a different type than an a TS enum or puts the descriptions in a separate object? Or perhaps I could parse the description comment strings out of the generated code file (although that feels a bit hacky)?

I'm using the following versions:

```
        "@graphql-codegen/cli": "^2.16.1",
        "@graphql-codegen/typescript": "^2.8.5",
        "@graphql-codegen/typescript-operations": "^2.5.10",
        "@graphql-codegen/typescript-react-apollo": "^3.3.7",
```

With the following config:

```
import { CodegenConfig } from '@graphql-codegen/cli';

const config: CodegenConfig = {
    schema: 'http://localhost:8000/graphql/',
    documents: ['src/gqlOperations/*'],
    generates: {
        './src/generated-types.ts': {
            plugins: ['typescript', 'typescript-operations', 'typescript-react-apollo'],
            config: {
                scalars: {
                    UUID: 'string',
                    DateTime: 'string',
                    JSON: 'unknown',
                },
            },
        },
    },
};

export default config;
```

**Comment by saihaj** [comment]
I suppose you can try to write a custom plugin where during the visit you can export strings as a type yourself?
